src/dialog/unix/SDL_portaldialog.c

```diff
@@ -305,6 +312,28 @@ void SDL_Portal_ShowFileDialogWithProperties(SDL_FileDialogType type, SDL_Dialog
     case SDL_FILEDIALOG_SAVEFILE:
         method = "SaveFile";
         method_title = SDL_GetStringProperty(props, SDL_PROP_FILE_DIALOG_TITLE_STRING, "Save File");
+        if (default_location) {
+            if (stat(default_location, &statbuf) == 0) {
+                save_file_existing = S_ISREG(statbuf.st_mode) || S_ISLNK(statbuf.st_mode);
+            } else if (errno == ENOENT) {
+                char *dirc = SDL_strdup(default_location);
+                if (dirc) {
+                    location_folder = SDL_strdup(dirname(dirc));
+                    SDL_free(dirc);
```

Comment:
location_folder might be pointing at memory in dirc, which should be freed after location_folder is used.

Reply:
What's the implication of memory allocation failing in this code? Should it be a fatal error, or is the fallback behavior correct?

Reply:
I feel the worst case scenario would be failing to show a dialogue, which I would think would return a DBus error.

Beyond that setting `current_folder` to whatever the developer passed in and letting the desktop portal interpret the value should at least present the end user with a save dialogue, even if it requires that user to navigate to somewhere more sensible.